### PR TITLE
Repackages consistency checker and adds support for proxy contracts

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

This PR:
- Repackages consistency checker into a first-class package
- Adds support for proxy contracts

Figure this PR is growing and this is a good break point.  Will follow this up with a PR for integration into origin-cron.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: infra/consistency/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "consistency",
```
Comment (tomlinton) [inline infra/consistency/package.json]: @origin/consistency?

File: infra/consistency/src/validators/ident.js [hunk 2]
```diff
@@ -145,24 +199,51 @@ function validateIPFSToDB(ipfsJson, dbRecord) {
...
-    log.error(`Did not find a matching record for address ${address}`)
+  if (owner.length < 1) {
+    log.error(
+      `Did not find a matching record for address ${address}(or proxy ${proxyAddress})`
+    )
     return false
-  } else if (records.length > 1) {
-    log.warn(`Found too many records for address ${address}`)
+  } else if (owner.length > 1) {
+    log.error(`Too many reords found(${owner.length})!`)
```
Comment (tomlinton) [inline infra/consistency/src/validators/ident.js]: Typo 👺

File: infra/consistency/src/validators/ident.js [hunk 3]
```diff
@@ -206,6 +291,21 @@ async function validateIdentities({
 }) {
   const identityEvents = contractsContext.identityEvents
   const identities = {}
+  const latestEvents = {}
+
+  const web3 = new Web3(
+    'https://eth-mainnet.alchemyapi.io/jsonrpc/FCA-3myPH5VFN8naOWyxDU6VkxelafK6'
```
Comment (tomlinton) [inline infra/consistency/src/validators/ident.js]: Should we make this configurable?

Comment (mikeshultz) [reply infra/consistency/src/validators/ident.js]: yeah, good catch.  Intended to do that and forgot all about it.  Thanks

**Comment by mikeshultz** [comment]
Integrated your feedback, thanks for the review!

> Consistency is important.

:|

**Pull request merged**